README.md

```diff
@@ -325,6 +326,191 @@ const commonProps = {
+    ...
+  />
+</div>
+```
+
+The props that `<ImgixProvider>` makes accessible can also be overridden by `<Imgix>` components. Any prop defined on the `<Imgix>` component will override the value set by the Provider.
+
+```jsx
+// inside App.jsx
+{
+  /*... */
+}
+<ImgixProvider imgixParams={{ ar: "16:9" }}>
+  <div className="intro-blurb">{/* ... */}s</div>
+  <div className="gallery">
+    <Imgix
+      imgixParams={{ ar: "4:2" }}
+      src="https://assets.imgix.net/examples/pione.jpg"
+    />
+    <Imgix src="https://sdk-test.imgix.net/ساندویچ.jpg" />
+  </div>
+</ImgixProvider>;
+{
+  /*... */
+}
+```
+
+So that the generated HTML looks something like this
+
+```html
+<div class="gallery">
+  <img src="https://assets.imgix.net/examples/pione.jpg?ar=4%3A2" ... />
+  <img
+    src="https://sdk-test.imgix.net/%D8%B3%D8%A7%D9%86%D8%AF%D9%88%DB%8C%DA%86.jpg?ar=16%3A9"
+    ...
+  />
+</div>
+```
+
+To remove a shared prop from an `<Imgix>` component, the same prop can be set to `undefined` on the component itself.
+
+```jsx
+// inside App.jsx
+{
+  /*... */
+}
+<ImgixProvider height={500}>
+  <div className="intro-blurb">{/* ... */}s</div>
+  <div className="gallery">
+    <Imgix src="https://assets.imgix.net/examples/pione.jpg" />
+    <Imgix height={undefined} src="https://sdk-test.imgix.net/ساندویچ.jpg" />
+  </div>
+</ImgixProvider>;
+{
+  /*... */
+}
+```
+
+So that the generated HTML looks something like this
+
+```html
+<div class="gallery">
+  <img src="https://assets.imgix.net/examples/pione.jpg?h=500" ... />
+  <img
+    src="https://sdk-test.imgix.net/%D8%B3%D8%A7%D9%86%D8%AF%D9%88%DB%8C%DA%86.jpg"
+    ...
+  />
+</div>
+```
+
+You cans nest `ImgixProvider` components to ensure that different consumers have different props.
+
+For example to give `Imgix` components different props from `Picture` components, you can next an `ImgixProvider` inside of another one.
+
+The nested Provider will change the Context for the `Picture` component, essentially removing their access to the shared props provided by the root `ImgixProvider`.
+
+```jsx
+import React from 'react'
+import Imgix, { ImgixProvider, Picture, Source } from "react-imgix";
+export default function simpleImage() {
+  return (
+    <div className="imgix-simple-api-example">
+      {/* there props will be accessible to all the imgix components */}
+      <ImgixProvider
+        domain="assets.imgix.net"
+        src="/examples/pione.jpg"
+        imgixParams={{ fit: "crop" }}
+      >
+        <Imgix width={200} height={500} src="/examples/pione.jpg" />
+        <Imgix domain="sdk-test.imgix.net" src="/ساندویچ.jpg" />
+        <ImgixProvider
+          {/* since we define a new provider here, the context is redefined for any child components */}
+        >
+          <Picture>
+            {/* imgixParams prop is no longer defined here */}
+            <Source
+              width={100}
+              htmlAttributes={{ media: "(min-width: 768px)" }}
+            />
```

Comment:
No change needed but just wanted to say [whoa](https://media.giphy.com/media/uPnKU86sFa2fm/giphy.gif)